### PR TITLE
Map & Set insertion order clarifications

### LINKED ISSUE #17327: Set&Map: What is "insertion order"?
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

### What specific section or headline is this issue about?

Description

### What information was incorrect, unhelpful, or incomplete?

> You can iterate through the elements of a set in insertion order.

> insertion order

### What did you expect to see?

What's "insertion order"?

What's "insertion" here?

"insertion" is "valid `.add(value)`"?
```js
const set = new Set;
set.add(1);
set.add(2);
set.delete(1);
set.add(3);
for (const value of set) {
  console.log(value);
}
```
output should be 23?

```js
const set = new Set;
set.add(1);
set.add(2);
set.add(1);
for (const value of set) {
  console.log(value);
}
```
12?

```js
const set = new Set;
set.add(1);
set.add(2);
set.delete(1)
set.add(1);
for (const value of set) {
  console.log(value);
}
```
21?

> The `add()` method appends a new element with a specified value to the end of a `Set` object.

> appends

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/set`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/set/index.md
* Last commit: https://github.com/mdn/content/commit/281e3b21178946c8301232a8eb50d11770ee8450
* Document last modified: 2022-02-18T09:53:31.000Z

</details>

### PR DESCRIPTION
**Pull request by rosahbruno**
#### Summary

Update documentation for `Set`s to clarify language around adds/appends/insert. Also updates the examples to reflect the info covered in the docs updates.

#### Motivation

Based on the comments in the related issue, the language around inserting values into a set had room for improvement.

#### Related issues

Fixes #17327 

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/set/index.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The **`Set`** object lets you store unique values of any type, whether {{Glossar
 
 ## Description
 
-`Set` objects are collections of values. You can iterate through the elements of a set in insertion order. A value in the `Set` **may only occur once**; it is unique in the `Set`'s collection.
+`Set` objects are collections of values. You can iterate through the elements of a set in insertion order. The insertion order corresponds to the order in which each element was inserted in to the `Set` by the `Set.prototype.add()` method. A value in the `Set` **may only occur once**; it is unique in the `Set`'s collection.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/set/index.md:21]:
```suggestion
`Set` objects are collections of values. A value in the `Set` **may only occur once**; it is unique in the `Set`'s collection. You can iterate through the elements of a set in insertion order. The _insertion order_ corresponds to the order in which each element was inserted into the set by the [`add()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add) method successfully (that is, there isn't an identical element already in the set).
```

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Thanks! The changes here look great. Could you port the same description to map as well, so we can close that issue after merging this PR?

**Comment by rosahbruno** [comment]
@Josh-Cena 
I moved over that description and updated it to use the `Map` methods rather than the `Set` ones

File: files/en-us/web/javascript/reference/global_objects/map/index.md [hunk 2]
```diff
@@ -21,9 +21,7 @@ either a key or a value.
 
 ## Description
 
-A `Map` object iterates its elements in insertion order — a
-{{jsxref("Statements/for...of", "for...of")}} loop returns an
-array of `[key, value]` for each iteration.
+`Map` objects are collections of values. A value in the `Map` **may only occur once**; it is unique in the `Map`'s collection. You can iterate through the elements of a map in insertion order. The _insertion order_ corresponds to the order in which each element was inserted into the map by the [`set()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/set) method successfully (that is, there wasn't an identical element already in the map when `set()` was called).
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/map/index.md:24]:
```suggestion
`Map` objects are collections of key-value pairs. A key in the `Map` **may only occur once**; it is unique in the `Map`'s collection. You can iterate through the elements of a map in insertion order. The _insertion order_ corresponds to the order in which each key-value pair was inserted into the map by the [`set()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/set) method successfully (that is, there wasn't an identical key already in the map when `set()` was called).
```

**Review by Josh-Cena — APPROVED** [review]
Thank you—the changes look great to me.

Congratulations on your first contribution here—welcome aboard!

**Pull request merged**